hal_st/stm32fxxx/FlashInternalHighCycleAreaStm.cpp

```diff
@@ -0,0 +1,163 @@
+#include DEVICE_HEADER
+#include "hal_st/stm32fxxx/FlashInternalHighCycleAreaStm.hpp"
+#include "hal_st/stm32fxxx/FlashInternalStmDetail.hpp"
+#include "infra/event/EventDispatcher.hpp"
+#include "infra/util/ByteRange.hpp"
+#include "infra/util/MemoryRange.hpp"
+#include <cstdint>
+
+namespace
+{
+    using ConstHalfWordRange = hal::FlashInternalHighCycleAreaStm::ConstHalfWordRange;
+
+    constexpr uint32_t fullSize{ FLASH_EDATA_SIZE };
+    constexpr uint32_t bankSize{ fullSize / 2 };
+    constexpr uint32_t bankSectorNumber{ FLASH_EDATA_SECTOR_NB };
+    constexpr uint32_t sectorSize{ bankSize / bankSectorNumber };
+    constexpr uint32_t bankSectorOffset{ FLASH_SECTOR_NB - bankSectorNumber };
+
+    template<uint32_t address>
+    constexpr auto ptr()
+    {
+        return reinterpret_cast<const uint16_t*>(address);
+    }
+
+    const ConstHalfWordRange bank1Range{ ptr<FLASH_EDATA_BASE>(), ptr<FLASH_EDATA_BASE + bankSize>() };
+    const ConstHalfWordRange bank2Range{ ptr<FLASH_EDATA_BASE + bankSize>(), ptr<FLASH_EDATA_BASE + fullSize>() };
+
+    ConstHalfWordRange GetBankRange(uint32_t bank)
+    {
+        switch (bank)
+        {
+            case FLASH_BANK_1:
+                return bank1Range;
+            case FLASH_BANK_2:
+                return bank2Range;
+            default:
+                std::abort();
+        }
+    }
+
+    template<typename _II, typename _OI>
+    constexpr _OI copy(_II first, _II last, _OI result)
```

Comment:
Identifiers starting with _ following by an uppercase character are reserved in C++
```suggestion
    template<typename II, typename OI>
    constexpr OI copy(II first, II last, OI result)
```